### PR TITLE
Improve DDR support to allow use of VS 2017

### PR DESCRIPTION
**Pull request by ChengJin01**
The change is to improve DDR support in script/code
so as to allow use of VS 2017 on Windows.

Signed-off-by: CHENGJin <jincheng@ca.ibm.com>

### CODE REVIEW

**Comment by ChengJin01** [comment]
The changes mainly corrrect the path to cl.exe in VS2017
(configure was genereated with omr/configure.ac via autoconf)

Reviewer: @keithc-ca , @pshipton
FYI: @DanHeidinga

**Comment by charliegracie** [comment]
@genie-omr build all

**Comment by charliegracie** [comment]
@keithc-ca what are your thoughts on this change?

File: configure.ac [hunk 1]
```diff
@@ -540,15 +540,12 @@ AS_IF([test "$enable_DDR" = yes -a "$cross_compiling" != "yes"],
...
-						[DIASDK_HOME="$(cygpath -a -w "$diasdk_dir")"],
-						[
-						diasdk_dir="$vc_bin_dir/../../../DIA SDK"
-						AS_IF([test -f "$diasdk_dir/include/dia2.h"],
-							[DIASDK_HOME="$(cygpath -a -w "$diasdk_dir")"]
-						)
-						]
+					diasdk_dir1="$vc_bin_dir/../../DIA SDK"
+					diasdk_dir2="$vc_bin_dir/../../../DIA SDK"
+					diasdk_dir3="$vc_bin_dir/../../../../../../../DIA SDK"
```
Comment (keithc-ca) [inline configure.ac]: For `diasdk_dir3`, please provide example absolute paths that are covered by this addition.

**Comment by keithc-ca** [comment]
The change to PdbScanner.cpp looks good, although I would have preferred the case labels in alphabetical order.

File: configure.ac (hunk 1 above)
Comment (ChengJin01) [reply configure.ac]: diasdk_dir3 is used to cover the case of VS2017 (works for both Professional and Community version). The reason is that the path to cl.exe changes on VS2017.
```
Administrator@JCHTORWIN101 /cygdrive/c/Program Files (x86)/Microsoft Visual Studio/2017/Professional
$ ls
 Common7      'DIA SDK'         JS         SDK           VB    VSSDK
 CoreCon       ImportProjects   Licenses  'Team Tools'   VC    Web
 DesignTools   JavaScript       MSBuild    TS            VC#   Xml

Administrator@JCHTORWIN101 /cygdrive/c/Program Files (x86)/Microsoft Visual Studio/2017/Professional
$ find . -name cl.exe
./VC/Tools/MSVC/14.14.26428/bin/Hostx64/x64/cl.exe
./VC/Tools/MSVC/14.14.26428/bin/Hostx64/x86/cl.exe
./VC/Tools/MSVC/14.14.26428/bin/Hostx86/x64/cl.exe
./VC/Tools/MSVC/14.14.26428/bin/Hostx86/x86/cl.exe
```
e.g.
In VS2017 Professional, the path to DIA SDK:  C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\DIA SDK while vc_bin_dir is C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.14.26428\bin\Hostx86\x64\cl.exe.  So, we need to step backwards from the path to cl.exe by 7 level to  reach the DIA SDK directory.

Comment (keithc-ca) [reply configure.ac]: Thanks for the explanation.

**Comment by charliegracie** [comment]
@genie-omr build all

**Comment by ChengJin01** [comment]
Already updated the title and the description of commit/PR.

**Comment by keithc-ca** [comment]
@ChengJin01 Can this PR title be changed to match the commit message: 'Improve DDR support to allow use of VS 2017', please?

**Comment by ChengJin01** [comment]
Already changed the PR's title.

**Pull request merged**